util/flags/manager.go

```diff
@@ -0,0 +1,154 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package flags implements the manager options utilities.
+package flags
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apiserver/pkg/server/routes"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/component-base/logs"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/filters"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+// ManagerOptions provides command line flags for manager options.
+type ManagerOptions struct {
+	// TLS Options
+	// These are used for the webhook and for the metrics server (the latter only if secure serving is used)
+
+	// TLSMinVersion is the field that stores the value of the --tls-min-version flag.
+	// For further details, please see the description of the flag.
+	TLSMinVersion string
+	// TLSCipherSuites is the field that stores the value of the --tls-cipher-suites flag.
+	// For further details, please see the description of the flag.
+	TLSCipherSuites []string
+
+	// Metrics Options
+	// These are used to configure the metrics server
+
+	// MetricsBindAddr is the field that stores the value of the --metrics-bind-addr flag.
+	// For further details, please see the description of the flag.
+	//
+	// Deprecated: This field will be removed in an upcoming release.
+	MetricsBindAddr string
+	// DiagnosticsAddress is the field that stores the value of the --diagnostics-address flag.
+	// For further details, please see the description of the flag.
+	DiagnosticsAddress string
+	// InsecureDiagnostics is the field that stores the value of the --insecure-diagnostics flag.
+	// For further details, please see the description of the flag.
+	InsecureDiagnostics bool
+}
+
+// AddManagerOptions adds the manager options flags to the flag set.
+func AddManagerOptions(fs *pflag.FlagSet, options *ManagerOptions) {
+	fs.StringVar(&options.TLSMinVersion, "tls-min-version", "VersionTLS12",
+		"The minimum TLS version in use by the webhook server and metrics server (the latter only if --insecure-diagnostics is not set to true).\n"+
+			fmt.Sprintf("Possible values are %s.", strings.Join(cliflag.TLSPossibleVersions(), ", ")),
+	)
+
+	tlsCipherPreferredValues := cliflag.PreferredTLSCipherNames()
+	tlsCipherInsecureValues := cliflag.InsecureTLSCipherNames()
+	fs.StringSliceVar(&options.TLSCipherSuites, "tls-cipher-suites", []string{},
+		"Comma-separated list of cipher suites for the webhook server and metrics server (the latter only if --insecure-diagnostics is not set to true). "+
+			"If omitted, the default Go cipher suites will be used. \n"+
+			"Preferred values: "+strings.Join(tlsCipherPreferredValues, ", ")+". \n"+
+			"Insecure values: "+strings.Join(tlsCipherInsecureValues, ", ")+".")
+
+	fs.StringVar(&options.MetricsBindAddr, "metrics-bind-addr", "",
```

Comment:
q: when can we actually remove this? (might be worth adding a comment)

Reply:
Good question. I'm not sure. I was even thinking about just dropping it, but it's too close to the release now.

Not sure if we should hard-code a release here, but maybe let's drop it for the next release? (fine for me to decide this now and update the description to reflect that)

Reply:
Also, slightly unrelated, but should we document the change on flags utils in the upgrade doc?

Reply:
Let's open an issue for removal of the flag in the next release (so it is tracked)

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/10937